### DISCUSSION TITLE
Lesson 7: Why do we pass mostRecentlyDeployed to FundMe?

### DISCUSSION THREAD

**Discussion by ShayanShamsi** [Cyfrin/foundry-full-course-cu#114] [Q&A] [answered]
Hello!

I am having trouble wrapping my head around why passing the mostRecentlyDeployed address to the FundMe constructor works in [script/Interactions.s.sol](https://github.com/Cyfrin/foundry-fund-me-f23/blob/main/script/Interactions.s.sol)? To my understanding, mostRecentlyDeployed contains the address of the most recent deployment whereas the [FundMe constructor](https://github.com/Cyfrin/foundry-fund-me-f23/blob/main/src/FundMe.sol) indicates that FundMe accepts the address of the relevant priceFeed. Why do we not pass priceFeed address to FundMe in the interactions script? More importantly, how does passing the address of the deployed FundMe contract itself work?

**Comment by ghost** [comment, marked as answer]
Hello @ShayanShamsi.

> To my understanding, mostRecentlyDeployed contains the address of the most recent deployment

You're correct.

> whereas the [FundMe constructor](https://github.com/Cyfrin/foundry-fund-me-f23/blob/main/src/FundMe.sol) indicates that FundMe accepts the address of the relevant priceFeed

Also correct.

> Why do we not pass priceFeed address to FundMe in the interactions script? 

Because we want to interact with an already deployed contract, not redeploy it.

> More importantly, how does passing the address of the deployed FundMe contract itself work?

First we get `mostRecentlyDeployed`, because we need the latest created instance of the contract, then we pass this address to `fundFundMe`. Inside this function we create a new **reference** to the `FundMe` contract by casting (or wrapping) the address to a contract, like this:

```
FundMe(payable(mostRecentlyDeployed)).fund{value: SEND_VALUE}();
```

This can only be done if the address points to a deployed contract with the same structure as the contract you are trying to reference, otherwise it will revert.

For example, I cannot cast `mostRecentlyDeployed` in lesson 7 to `SimpleStorage(mostRecentlyDeployed)`, because the address points to a `FundMe` contract.

After creating the reference you can interact with the contract to call its methods, in this case the `fund` method.

**Reply by cromewar** [reply]
Amazing answer @pacelliv, marking this as answer.

**Reply by ShayanShamsi** [reply]
Thankyou @pacelliv ! Made everything super clear for me.
Earlier, I confused casting the address to a contract with the deployment of a new contract. I should have seen that the `new` keyword was not present before `FundMe` which is an indication of casting rather than instantiation.
Thanks again!
